cherab/tools/observers/bolometry.py

```diff
@@ -276,229 +227,151 @@ def __init__(self, detector_id, centre_point, basis_x, dx, basis_y, dy, slit,
-            pixel_samples=1000, x_width=dx, y_width=dy,
-            spectral_bins=1, parent=self, name=detector_id, quiet=True,
-            render_engine=render_engine
-        )
-
-        # NOTE - the los observer may not be normal to the surface, in which case calculate an extra relative transform
-        if self._normal_vec.dot(self._foil_to_slit_vec) != 1.0:
-            relative_foil_to_slit_vec = self._foil_to_slit_vec.transform(self.to_local())
-            relative_rotation = rotate_basis(relative_foil_to_slit_vec, Vector3D(1, 0, 0))
-            self._los_observer.transform = relative_rotation
-            self._los_cos_theta = self._normal_vec.dot(self._foil_to_slit_vec)
+        if self.units == "Power":
+            pipeline = PowerPipeline0D(accumulate=accumulate)
+        elif self.units == "Radiance":
+            pipeline = RadiancePipeline0D(accumulate=accumulate)
         else:
-            self._los_cos_theta = 1.0
-
-        # initialise sensitivity values
-        self._los_radiance_sensitivity = None
-        self._volume_radiance_sensitivity = None
-        self._volume_power_sensitivity = None
-        self._etendue = None
-        self._etendue_error = None
-
-    def __getstate__(self, serialisation_format=None):
-
-        state = {
-            'CHERAB_Object_Type': 'BolometerFoil',
-            'Version': 1,
-            'Detector_ID': self.detector_id,
-            'centre_point': self.centre_point.__getstate__(),
-            'basis_x': self._basis_x.__getstate__(),
-            'basis_y': self._basis_y.__getstate__(),
-            'dx': self.dx,
-            'dy': self.dy,
-            'slit_id': self._slit.slit_id,
-        }
-
-        # add extra data if saving as binary
-        if serialisation_format in ['.pickle', '.sav'] and self._los_radiance_sensitivity is not None:
-            state['los_radiance_sensitivity'] = self._los_radiance_sensitivity.__getstate__()
-            state['volume_radiance_sensitivity'] = self._volume_radiance_sensitivity.__getstate__()
-            state['volume_power_sensitivity'] = self._volume_power_sensitivity.__getstate__()
-            state['etendue'] = self._etendue
-            state['etendue_error'] = self._etendue_error
-
-        return state
+            raise ValueError("The units argument of BolometerFoil must be one of 'Power' or 'Radiance'.")
+
+        super().__init__([slit.target], targetted_path_prob=1.0,
+                         pipelines=[pipeline],
+                         pixel_samples=1000, x_width=dx, y_width=dy, spectral_bins=1, quiet=True,
+                         parent=parent, transform=translation * rotation, name=detector_id)
+
+    def __repr__(self):
+        """Returns a string representation of this BolometerFoil object."""
+        return "<BolometerFoil - " + self.name + ">"
 
     @property
     def centre_point(self):
-        return self._centre_point
+        return Point3D(0, 0, 0).transform(self.to_root())
```

Comment:
I like this. It really makes clear what the transforms are doing. And will be good for catching erroneous transforms (like the basis_x vs basis_y rotation bug we caught a couple of weeks ago).

Reply:
agreed